src/docfx/lib/log/SourceInfo.cs

```diff
@@ -16,7 +16,7 @@ public sealed class SourceInfo<T> : ISourceInfo
 
         object ISourceInfo.Value => Value;
 
-        public SourceInfo(T value, string file, Range range)
+        public SourceInfo(T value, Range range = default, string file = null)
```

Comment:
As `SourceInfo` has `file` already, seems we don't need `Document file` as a parameter in most methods of `errors.cs`.

Reply:
will clean this up after we merge #4311